lib/Driver/Driver.cpp

```diff
@@ -2757,6 +2767,33 @@ void Driver::chooseRemappingOutputPath(Compilation &C,
   }
 }
 
+void Driver::chooseTextualInterfacePath(Compilation &C, const JobAction *JA,
+                                        StringRef workingDirectory,
+                                        llvm::SmallString<128> &buffer,
+                                        CommandOutput *output) const {
+  switch (C.getOutputInfo().CompilerMode) {
+  case OutputInfo::Mode::StandardCompile:
+  case OutputInfo::Mode::BatchModeCompile:
+    if (!isa<MergeModuleJobAction>(JA))
+      return;
```

Comment:
I'm not super keen on this pattern of silently not-doing what the function's name says it does, when not matching certain conditions not-expressed in the signature; it means that reading the caller, one's assumption that "a textual interface output name has been set" is not correct. But I know it matches other filename-choosing functions. Do you feel like there's an easy way to tidy it up a bit?